pkgs/tools/filesystems/bcachefs-tools/default.nix

```diff
@@ -65,20 +57,27 @@ stdenv.mkDerivation (finalAttrs: {
     udev
   ] ++ lib.optional fuseSupport fuse3;
 
-  # FIXME: Try enabling this once the default linux kernel is at least 6.7
-  doCheck = false; # needs bcachefs module loaded on builder
-  checkFlags = [ "BCACHEFS_TEST_USE_VALGRIND=no" ];
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    src = finalAttrs.src;
+    hash = "sha256-RsRz/nb8L+pL1U4l6RnvqeDFddPvcBFH4wdV7G60pxA=";
+  };
 
   makeFlags = [
-    "PREFIX=${placeholder "out"}"
+    "DESTDIR=${placeholder "out"}"
+    "PREFIX="
     "VERSION=${finalAttrs.version}"
     "INITRAMFS_DIR=${placeholder "out"}/etc/initramfs-tools"
-    "BCACHEFS_FUSE=${toString fuseSupport}"
-  ];
+  ] ++ lib.optional fuseSupport "BCACHEFS_FUSE=1";
+
+
+  # FIXME: Try enabling this once the default linux kernel is at least 6.7
+  doCheck = false; # needs bcachefs module loaded on builder
 
   preCheck = lib.optionalString (!fuseSupport) ''
     rm tests/test_fuse.py
   '';
+  checkPhase = "./target/release/bcachefs version";
```

Comment:
What's the point of setting this if `doCheck` is `false`?

Reply:
I have added this check for future, when default LTS kernel becomes > 6.7. Then we can enable doCheck = true and that will run this script

Reply:
But that overrides the default checkPhase, so now it won't run any of the tests it would have run before.

What's the motivation for testing that `bcachefs version` works?  It feels like not a very meaningful test unless I'm missing something.

Reply:
Oh, I thought I would add a step to checkPhase not override it. While `bcachefs --version` is not an extensive check, it could show if there is a fatal runtime error due to missing libraries/packages.

I'll revert this so default checkPhase is used, if enabled that is.